osu.Game/Screens/Play/Player.cs

```diff
@@ -241,6 +246,16 @@ private void addOverlayComponents(Container target, WorkingBeatmap working)
             });
         }
 
+        private readonly Bindable<float> dimOffset = new Bindable<float>();
+
+        private void updateUserDimState()
+        {
+            const float light_amount = 0.3f;
```

Comment:
can all this please be moved inside `UserDimContainer`, exposing only a single `bool ApplyBreakLightening` or similar.

the goal is to add as few lines as possible to Player.cs, since the class is already quite large.

Reply:
even though `UserDimContainer` seems out of scope for this case but sure.

Reply:
what do you mean by out of scope?

Reply:
not in the right place, `UserDimContainer` is just meant to hold some user-configured dim and blur levels of the container, but I suppose it's fine for the sake of making `Player` more straightforward.

One thing to consider is to move the `UserDimContainer` somewhere else (probably inside `osu.Game.Screens.Play`), just my opinion.

Reply:
Correct me if I'm wrong, but `UserDimContainer` is *only* used in Player, right? If there was another usage I would understand.

Reply:
yeah, but I was talking from a placement and code quality perspective. it's inside `osu.Game.Graphics.Containers`.

Though I've just realized there are storyboard and video references inside the container, unsure of whether this is out-of-scope or not anymore...

Reply:
moving `UserDimContainer` is fine.